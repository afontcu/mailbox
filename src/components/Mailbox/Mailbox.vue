<template>
  <div class="Mailbox">
    <h1 class="Mailbox__title">Mailbox</h1>
    <div class="Mailbox__grid">
      <mailbox-message-list :messageList="messages" />
      <mailbox-message-content v-bind="selectedMessage" />
    </div>
  </div>
</template>

<script>
import { mapGetters } from 'vuex'
import MailboxMessageList from '@/components/MailboxMessageList'
import MailboxMessageContent from '@/components/MailboxMessageContent'

export default {
  name: 'Mailbox',
  components: {
    MailboxMessageList,
    MailboxMessageContent,
  },
  computed: {
    ...mapGetters(['messages', 'selectedMessage']),
  },
}
</script>

<style>
.Mailbox {
  max-width: 768px;
  margin: 0 auto;
  padding: 0 var(--global-spacing);
}

.Mailbox__title {
  margin-bottom: var(--global-spacing-large);
  padding-bottom: var(--global-spacing-large);

  text-align: center;
  font-weight: bold;
}

.Mailbox__grid {
  display: grid;
  grid-template-columns: 250px 1fr;
  grid-gap: var(--global-spacing);
}
</style>
