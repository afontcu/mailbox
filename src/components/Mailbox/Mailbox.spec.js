import Mailbox from './index.js'
import { mount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'

const localVue = createLocalVue()
localVue.use(Vuex)

describe('Mailbox.vue', () => {
  let store, getters

  beforeEach(() => {
    getters = {
      messages: jest.fn(() => []),
      selectedMessage: jest.fn(() => ({})),
      unreadMessagesCount: () => 3,
    }
    store = new Vuex.Store({
      getters,
    })
  })

  it('is a component', () => {
    const wrapper = mount(Mailbox, { store, localVue })
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('renders main node', () => {
    const wrapper = mount(Mailbox, { store, localVue })
    expect(wrapper.classes()).toContain('Mailbox')
  })

  it('renders a snapshot', () => {
    const wrapper = mount(Mailbox, { store, localVue })
    expect(wrapper.html()).toMatchSnapshot('default')
  })
})
