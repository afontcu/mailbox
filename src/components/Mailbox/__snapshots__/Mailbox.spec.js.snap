// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mailbox.vue renders a snapshot: default 1`] = `
<div class="Mailbox">
  <h1 class="Mailbox__title">Mailbox</h1>
  <div class="Mailbox__grid">
    <aside class="MailboxMessageList">
      <p class="MailboxMessageList__title">
        3 unread
        messages
      </p>
    </aside>
    <main class="MailboxMessageContent">
      <p class="MailboxMessageContent__no-message">
        Select a message to display its content.
      </p>
    </main>
  </div>
</div>
`;
